### ISSUE TITLE
Devpi couldn't use existing directory if switched from master to replica

### ISSUE THREAD

**Issue by orangetreee** [devpi/devpi#852]
My devpi version:
devpi (2.2.0)
devpi-client (5.2.2)
devpi-common (3.6.0)
devpi-server (6.2.0)
devpi-web (4.0.8)

Hi, I want to switch a replica to master, but get error like this:

fatal: The path 'dir/pip-repo/pypi' already contains devpi-server data.

The command I use:
 devpi-init --role replica --master-url http://xxx:xxx --serverdir dir/pip-repo/pypi --no-root-pypi --root-passwd 23dg@sQ!

Anyone know how to deal with it?

**Comment by fschulze** [comment]
If you want to switch the role, you don't need ``devpi-init``.
1. Stop the replica
2. Use ``devpi-fsck --serverdir dir/pip-repo/pypi`` to check that your replica is complete and consistent
3. Change your ``devpi-server`` call to use ``--role master --serverdir dir/pip-repo/pypi`` instead of ``--role replica --serverdir dir/pip-repo/pypi`` and remove ``--master-url http://...``
4. Consider changing your root password if ``23dg@sQ!`` is your actual password

**Comment by orangetreee** [comment]
No, thery are not in sync and serial is different. The replica's serial is still the old one. I tried to delete the .nodeinfo and run devpi-init again. But it doesn't change.

<img width="594" alt="截屏2021-10-08 下午6 39 12" src="https://user-images.githubusercontent.com/47046449/136544411-88fc907f-0af5-4b33-a512-1818bb26ff0d.png">
<img width="625" alt="截屏2021-10-08 下午6 39 02" src="https://user-images.githubusercontent.com/47046449/136544423-f485b0e8-9f7c-4b91-87b5-254ffe620431.png">
